lib/reversetunnel/localsite.go

```diff
@@ -217,24 +217,28 @@ func (s *localSite) GetLastConnected() time.Time {
 	return s.clock.Now()
 }
 
-func (s *localSite) DialAuthServer() (net.Conn, error) {
+func (s *localSite) DialAuthServer(params DialParams) (net.Conn, bool, error) {
 	if len(s.authServers) == 0 {
-		return nil, trace.ConnectionProblem(nil, "no auth servers available")
+		return nil, false, trace.ConnectionProblem(nil, "no auth servers available")
 	}
 
 	addr := utils.ChooseRandomString(s.authServers)
 	conn, err := net.DialTimeout("tcp", addr, apidefaults.DefaultDialTimeout)
 	if err != nil {
-		return nil, trace.ConnectionProblem(err, "unable to connect to auth server")
+		return nil, false, trace.ConnectionProblem(err, "unable to connect to auth server")
 	}
 
-	return conn, nil
+	if err := s.sendSignedPROXYHeader(params, conn, false, false); err != nil {
```

Comment:
Since `dialWithAgent` doesn't return actual underlying connection to node we need to send header inside. But to be honest, I experimented with it being up the chain and I think `localSite` is actually a good place for this.